### PR TITLE
feat: add view collection

### LINKED ISSUE #36: [Subtask]: Asset management: View collections

### PR DESCRIPTION
**Pull request by cpl121**
I have needed to add 2 dependencies in the package.json
- metaplex-foundation/js to find all nfts by candy machine id
- swiper to add a carousel and show the tokens

The tokens that are displayed on the view page will only be seen when they have been minted
I have also added a token view in inspect candy machine by preloading the cache file, because I think it is interesting to know which tokens have been uploaded to the candy machine

Closes #36

### CODE REVIEW

File: lib/candy-machine/constants.ts [hunk 1]
```diff
@@ -49,6 +50,27 @@ export const CONFIG_ARRAY_START_V2 =
     32 +
     1 // gatekeeper
 
+export const CONFIG_ARRAY_START =
```
Comment (begonaalvarezd) [inline lib/candy-machine/constants.ts:53]: What do you need this for?

File: lib/candy-machine/constants.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { PublicKey } from '@solana/web3.js'
 export const CANDY_MACHINE_PROGRAM_V2_ID = new PublicKey('cndy3Z4yapfJBmL3ShUp5exZKqR3z33thTzeNMm2gRZ')
 export const SPL_ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID = new PublicKey('ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL')
 export const TOKEN_PROGRAM_ID = new PublicKey('TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA')
+export const TOKEN_METADATA_PROGRAM_ID = new PublicKey('metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s')
```
Comment (begonaalvarezd) [inline lib/candy-machine/constants.ts:6]: What do you need this for?

File: lib/candy-machine/constants.ts [hunk 3]
```diff
@@ -49,6 +50,27 @@ export const CONFIG_ARRAY_START_V2 =
...
+    1 + //is mutable
+    1 + // retain authority
+    4 // max number of lines;
+
+export const MAX_DATA_SIZE =
+    4 + MAX_NAME_LENGTH + 4 + MAX_SYMBOL_LENGTH + 4 + MAX_URI_LENGTH + 2 + 1 + 4 + MAX_CREATOR_LIMIT * MAX_CREATOR_LEN
+export const MAX_METADATA_LEN = 1 + 32 + 32 + MAX_DATA_SIZE + 1 + 1 + 9 + 172
+export const CREATOR_ARRAY_START =
+    1 + 32 + 32 + 4 + MAX_NAME_LENGTH + 4 + MAX_URI_LENGTH + 4 + MAX_SYMBOL_LENGTH + 2 + 1 + 4
+
```
Comment (begonaalvarezd) [inline lib/candy-machine/constants.ts:73]: What do you need any of theses consts for?

File: lib/candy-machine/constants.ts (hunk 1 above)
Comment (cpl121) [reply lib/candy-machine/constants.ts:53]: I used them before using the metaplex-foundation dependency and I forgot to delete them, now I correct it

File: lib/candy-machine/view/interfaces.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export interface Token {
```
[the remark this answers is not in the document]
Comment (begonaalvarezd) [reply lib/candy-machine/view/interfaces.ts:1]: move to `lib/nft`

File: components/layout/Carousel.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import Swiper, { Navigation } from 'swiper'
+import { FC } from 'react'
+import { Token } from 'lib/candy-machine/view/interfaces'
+
+const Carousel: FC<{
+    token: Token[]
```
Comment (begonaalvarezd) [inline components/layout/Carousel.tsx:6]: Carousel should be agnostic to NFTs, Tokens or whatever. The Carousel only cares about assets and assets titles

File: pages/view-candy-machine/[id].tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        const result = []
+        for (let i = 0; i < nfts.length; i++) {
+            let fetchImage = await fetch(nfts[i].uri)
+            let imageData = await fetchImage.json()
+            let tokenData = {
+                name: nfts[i].name,
+                imageLink: imageData.image,
+            }
+            result.push(tokenData)
+        }
```
Comment (begonaalvarezd) [inline pages/view-candy-machine/[id].tsx:38]: This can be moved to `lib/nfts/actions.ts`

File: pages/view-candy-machine/[id].tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+import { useWallet } from '@solana/wallet-adapter-react'
+import { useEffect, useState } from 'react'
+import { PublicKey } from '@solana/web3.js'
+import { useConnection } from '@solana/wallet-adapter-react'
+import { Token } from 'lib/candy-machine/view/interfaces'
+import { Metaplex } from '@metaplex-foundation/js'
+
+const ViewCandyMachine: NextPage = () => {
+    const router = useRouter()
+    const account = router.query.id
```
Comment (begonaalvarezd) [inline pages/view-candy-machine/[id].tsx:14]: rename this to something related to candy machine account/address or something

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Very nice work! 
Added some comments 🙏🏼

**Review by aspnxdd — CHANGES_REQUESTED** [review]

**Pull request merged**